From 1cb210981d2baf325aaab36cdd84afa40c775f00 Mon Oct 14 08:23:08 2024
From: Your Name <your.email@example.com>
Date: Mon Oct 14 08:23:08 UTC 2024
Subject: [PATCH 1/1] Your patch subject

Change-Id: 1cb210981d2baf325aaab36cdd84afa40c775f00
---
diff --git a/Android.bp b/Android.bp
index 683c24c..b9e2084 100644
--- a/Android.bp
+++ b/Android.bp
@@ -129,67 +129,3 @@ prebuilt_etc {
},
}

-// Defining prebuilt libraries for the compatibility of old vendor modules
-cc_prebuilt_library_shared {
-    name: "libprotobuf-cpp-lite-vendorcompat",
-    stem: "libprotobuf-cpp-lite",
-    vendor: true,
-    // These are already stripped, and restripping them just issues diagnostics.
-    strip: {
-        none: true,
-    },
-    shared_libs: [
-        "libc++",
-        "libc",
-        "libdl",
-        "liblog",
-        "libm",
-        "libz",
-    ],
-    target: {
-        android_arm64: {
-            srcs: ["arm64/arch-arm64-armv8-a/shared/vndk-core/libprotobuf-cpp-lite.so"],
-        },
-        android_arm: {
-            srcs: ["arm/arch-arm-armv7-a-neon/shared/vndk-core/libprotobuf-cpp-lite.so"],
-        },
-        android_x86_64: {
-            srcs: ["x86_64/arch-x86_64-x86_64/shared/vndk-core/libprotobuf-cpp-lite.so"],
-        },
-        android_x86: {
-            srcs: ["x86/arch-x86-x86/shared/vndk-core/libprotobuf-cpp-lite.so"],
-        },
-    },
-}
-
-cc_prebuilt_library_shared {
-    name: "libprotobuf-cpp-full-vendorcompat",
-    stem: "libprotobuf-cpp-full",
-    vendor: true,
-    // These are already stripped, and restripping them just issues diagnostics.
-    strip: {
-        none: true,
-    },
-    shared_libs: [
-        "libc++",
-        "libc",
-        "libdl",
-        "liblog",
-        "libm",
-        "libz",
-    ],
-    target: {
-        android_arm64: {
-            srcs: ["arm64/arch-arm64-armv8-a/shared/vndk-core/libprotobuf-cpp-full.so"],
-        },
-        android_arm: {
-            srcs: ["arm/arch-arm-armv7-a-neon/shared/vndk-core/libprotobuf-cpp-full.so"],
-        },
-        android_x86_64: {
-            srcs: ["x86_64/arch-x86_64-x86_64/shared/vndk-core/libprotobuf-cpp-full.so"],
-        },
-        android_x86: {
-            srcs: ["x86/arch-x86-x86/shared/vndk-core/libprotobuf-cpp-full.so"],
-        },
-    },
-}
