diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 9cc9118..816a999 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -19,27 +19,6 @@ bootstrap_go_package {
 
 pixelage_generator {
     name: "generated_kernel_includes",
-
-    // The headers make command
-    cmd: "$(PATH_OVERRIDE_SOONG) $(KERNEL_MAKE_CMD) $(KERNEL_MAKE_FLAGS) -C $(TARGET_KERNEL_SOURCE) O=$(KERNEL_BUILD_OUT_PREFIX)$(genDir) ARCH=$(KERNEL_ARCH) $(KERNEL_CROSS_COMPILE) headers_install && vendor/pixelage/tools/clean_headers.sh $(KERNEL_BUILD_OUT_PREFIX)$(genDir)",
-
-    // Directories that can be imported by a cc_* module generated_headers property
-    export_include_dirs: [
-        "usr/audio/include/uapi",
-        "usr/include",
-        "usr/include/audio",
-        "usr/include/audio/include/uapi",
-        "usr/techpack/audio/include",
-    ],
-
-    // Sources for dependency tracking
-    dep_root: "$(TARGET_KERNEL_SOURCE)",
-    dep_files: [
-        "Makefile",
-        "include/**/*",
-        "arch/$(KERNEL_ARCH)/include/**/*",
-        "techpack/audio/include/**/*",
-    ],
 }
 
 cc_defaults {
diff --git a/config/common.mk b/config/common.mk
index fd12ed2..df0eda7 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -27,16 +27,8 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 endif
 endif
 
-ifeq ($(TARGET_BUILD_VARIANT),eng)
-# Disable ADB authentication
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=0
-else
-# Enable ADB authentication
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=1
-
 # Disable extra StrictMode features on all non-engineering builds
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += persist.sys.strictmode.disable=true
-endif
 
 # Backup Tool
 PRODUCT_COPY_FILES += \
diff --git a/config/version.mk b/config/version.mk
index fa096cd..3a63ccc 100644
--- a/config/version.mk
+++ b/config/version.mk
@@ -17,7 +17,6 @@ PIXELAGE_VERSION := ProjectPixelage-v$(PIXELAGE_DISPLAY_VERSION)-$(PIXELAGE_BUIL
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.pixelage.build.date=$(BUILD_DATE) \
     ro.pixelage.build.version=$(PIXELAGE_DISPLAY_VERSION) \
-    ro.pixelage.device=$(PIXELAGE_BUILD) \
     ro.pixelage.fingerprint=$(ROM_FINGERPRINT) \
     ro.pixelage.releasetype=$(PIXELAGE_BUILDTYPE) \
     ro.pixelage.version=$(PIXELAGE_VERSION) \
